### ISSUE TITLE
TripoSR bad geometry generation

### ISSUE THREAD

**Issue by kakachiex2** [MrForExample/ComfyUI-3D-Pack#118]
Suddenly my workflow shows this mesh; I create a setup with the Flowty TripoSR to see if it do the same but work ok.
![Screenshot 2024-03-19 090420](https://github.com/MrForExample/ComfyUI-3D-Pack/assets/1762112/d7ac9ac8-87b4-4aab-8f85-046325af7baf)

**Comment by MrForExample** [comment]
> Suddenly my workflow shows this mesh; I create a setup with the Flowty TripoSR to see if it do the same but work ok. ![Screenshot 2024-03-19 090420](https://private-user-images.githubusercontent.com/1762112/314063791-d7ac9ac8-87b4-4aab-8f85-046325af7baf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA4ODgxMTcsIm5iZiI6MTcxMDg4NzgxNywicGF0aCI6Ii8xNzYyMTEyLzMxNDA2Mzc5MS1kN2FjOWFjOC04N2I0LTRhYWItOGY4NS0wNDYzMjVhZjdiYWYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMxOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMTlUMjIzNjU3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ODM0YTdiZTFhZGY5OWQwYWM1NDQ2NjM0NWVlNDRhYTRmNjAwODc3YmZjYWIzMmQzZTUzYzVmYjFlMWFhMTZjNCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.UMGuF16eHsBu5F-fD776QoLSk4ai90UPkd_n9QuhMIg)

Yeah, about this problem, you probably omitted invert mask node somewhere, since by default the mask as a image is inverted when you load the image (By default object part of mask is black and background is white, and you want the opposite before feeding it into CRM's nodes)

**Comment by MrForExample** [comment]
> I have small feature request I like to resize the 3d viewer to match my workflow proportion and keep good organization but wen loading the workflow the viewer go back to its default size. Another feature is to connect the viewer directly from the Switch Mesh Axis not always want to save the mesh.

Sure, sounds reasonable enough, I'll see what I can do 👍

**Comment by kakachiex2** [comment]
It has the invert mask if you see the screen, it says get invert mask, I tested with two mode normal mask and invert mask and keep doing the same.

**Comment by kakachiex2** [comment]
@MrForExample I create a new TripoSR workflow from start with the same configuration and my first TripoSR workflow old one.
- the old workflow render the mesh correctly
- the second new workflow it don't render the mesh correctly what is happening

Old Workflow :: 
![Screenshot 2024-03-22 140131](https://github.com/MrForExample/ComfyUI-3D-Pack/assets/1762112/26bc9396-892c-43f3-81ea-d1941c501cd5)

New Workflow :: 
![Screenshot 2024-03-22 140151](https://github.com/MrForExample/ComfyUI-3D-Pack/assets/1762112/13da05ff-e3d8-4e36-a959-333d0552fe36)

**Comment by kakachiex2** [comment]
I create a new test workflow for TripoSR and know I'm getting this new errors.

Error occurred when executing [Comfy3D] TripoSR:

Expected all tensors to be on the same device, but found at least two devices, cpu and cuda:0! (when checking argument for argument tensors in method wrapper_CUDA_cat)

File "K:\ComfyUI\ComfyUI_Ex\ComfyUI\execution.py", line 151, in recursive_execute
output_data, output_ui = get_output_data(obj, input_data_all)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "K:\ComfyUI\ComfyUI_Ex\ComfyUI\execution.py", line 81, in get_output_data
return_values = map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "K:\ComfyUI\ComfyUI_Ex\ComfyUI\execution.py", line 74, in map_node_over_list
results.append(getattr(obj, func)(**slice_dict(input_data_all, i)))
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "K:\ComfyUI\ComfyUI_Ex\python_miniconda_env\ComfyUI\Lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
return func(*args, **kwargs)
^^^^^^^^^^^^^^^^^^^^^
File "K:\ComfyUI\ComfyUI_Ex\ComfyUI\custom_nodes\ComfyUI-3D-Pack\nodes.py", line 1653, in run_TSR
image = torch.cat((image, mask), dim=2).detach().cpu().numpy()
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

**Comment by kakachiex2** [comment]
Ok I found the reason I was using BIREfNet to extract the mask and after processing the invert mask to TripoSR it gives these errors but know I'm using BRIA Remove background for the mask and don't throw these errors but keep generating bad mesh.

![Screenshot 2024-03-22 151929](https://github.com/MrForExample/ComfyUI-3D-Pack/assets/1762112/9dfcc297-5da1-4a97-9b3b-4a7948497be0)

**Comment by kakachiex2** [comment]
After several test looks like something change and your TripSR is not taking into account the inverted mask it only works with normal mask and the mesh looks les detailed.
Here Im using not inverted mask and it works : 

![Screenshot 2024-03-22 163149](https://github.com/MrForExample/ComfyUI-3D-Pack/assets/1762112/04c16a3a-719b-41c9-93d6-9e36dece61bd)
